src/main/java/org/marketdesignresearch/mechlib/core/bidder/newstrategy/truthful/BoundRandomHelper.java

```diff
@@ -0,0 +1,83 @@
+package org.marketdesignresearch.mechlib.core.bidder.newstrategy.truthful;
+
+import java.math.BigDecimal;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+import java.util.WeakHashMap;
+
+import org.marketdesignresearch.mechlib.core.Bundle;
+import org.marketdesignresearch.mechlib.core.Domain;
+import org.marketdesignresearch.mechlib.core.bid.bundle.BundleBoundValuePair;
+import org.marketdesignresearch.mechlib.core.bidder.Bidder;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class BoundRandomHelper {
+	
+	public static BigDecimal DEFAULT_STD_FOR_BOUNDS_GENERATION = BigDecimal.valueOf(0.5);
+	
+	// To not prevent garbage collection of a model if it is not used anymore
+	private static Map<Bidder,Random> randomMap = new WeakHashMap<>();
+	private static Map<Bidder,BigDecimal> stdMap = new WeakHashMap<>();

```

Comment:
In the version with Bounds bidders have many truthful strategies. In order to determine the strategy the bidder (or his strategy) needs to have a Random instance that can be parametrized for reproducability. 
Instead of adding this functionallity to the bidder or his strategy I added a class here that holds all bidder parameters for the Bound strategies (the stdMap holds the standard deviation for the bound generation, i.e. determining how wide upper and lower bounds are in expectation). 

Adding these parameters to the strategy would maybe the cleanest way, but then you would need to set the strategy for each bidder manually, if you wan't to fix or change this parameters.

What do you think about this issue? Note that the parameters here are set by domain, so you can run multiple auctions in the same execution environment with different parameters.

Reply:
I see such a helper class as an elegant solution to this, at least for now. If this issue pops up again in other use cases regarding the strategies, we may want to have a more principled solution for it, but I think for this use case alone, that's a good way to solve it.
It would be great if you could document the class and methods a bit, such that it becomes obvious why this class is here and what its purpose is.

Reply:
Thx for your toughts. Indeed this helper would be a solution for now. The only problem I see is that the auction cannot rollback at the moment and reproduce the same output if there is some randomness in place. 

1. Therefore the Random object used needs to depend on the current AuctionRound
2. Moving this randomness to the bidder would be the cleanest way for me. However, then we need some nice mechanism to set all the bidders initial seeds which generates alot of overhead.
3. Therefore I suggest moving the Random object to the Auction directly. The AuctionPhase (the mechanism) and the Bidder (via a StrategyContext) might access and use the same round dependent Random object. Therefore we just need one seed for the domain and one seed for the auction and are almost good.

The only downside on this solution is that it is hard to remove or add some component (like generating the bounds) and still get e.g. the same Random initial bids to have both auctions comparable. But personally breaking things here would be fine for me as soon as the results are reproducable.